Chummer/Classes/clsCustomDataDirectoryInfo.cs

```diff
@@ -0,0 +1,475 @@
+
+        /// <summary>
+        /// The path to the Custom Data Directory
+        /// </summary>
+        public string Path { get; }
+
+        /// <summary>
+        /// The version of the custom data directory
+        /// </summary>
+        public int Version => _version;
+
+
+        // /// <summary>
+        // /// The Sha512 Hash of all non manifest.xml files in the directory
+        // /// </summary>
+        // public int Hash { get; private set; }
+
+        /// <summary>
+        /// A list of all dependencies each formatted as Tuple(guid, str name, int version, bool isMinimumVersion).
+        /// </summary>
+        public IReadOnlyList<Tuple<Guid, string, int, bool>> DependenciesList => _lstDependencies;
+
+        /// <summary>
+        /// A list of all exclusivities each formatted as Tuple(int guid, str name, int version, bool ignoreVersion).
+        /// </summary>
+        public IReadOnlyList<Tuple<Guid, string, int, bool>> ExclusivitiesList => _lstExclusivities;
+
+        /// <summary>
+        /// A Dictionary containing all Descriptions, which uses the language code as key
+        /// </summary>
+        public IDictionary<string, string> DescriptionDictionary { get; } = new Dictionary<string, string>();
+
+        /// <summary>
+        /// The description to display
+        /// </summary>
+        public string DisplayDescription { get; private set; }
+
+        /// <summary>
+        /// A Dictionary, that uses the author name as key and provides a bool if he is the main author
+        /// </summary>
+        public IDictionary<string, bool> AuthorDictionary { get; } = new Dictionary<string, bool>();
+
+        /// <summary>
+        /// A string containing all Authors formatted as Author(main), Author2 
+        /// </summary>
+        public string DisplayAuthors { get; private set; }
+
+        public Guid Guid => _guid;
+        #endregion
+
+        #region Interface Implementations and Operators
+        public int CompareTo(object obj)
+        {
+            if (obj is CustomDataDirectoryInfo objOtherDirectoryInfo)
+                return CompareTo(objOtherDirectoryInfo);
+            return 1;
+        }
+
+        public int CompareTo(CustomDataDirectoryInfo other)
+        {
+            int intReturn = string.Compare(Name, other.Name, StringComparison.Ordinal);
+            if (intReturn == 0)
+                intReturn = string.Compare(Path, other.Path, StringComparison.Ordinal);
+            return intReturn;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is CustomDataDirectoryInfo objOther)
+                return Equals(objOther);
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            return new { Name, Path }.GetHashCode();
+        }
+
+        public bool Equals(CustomDataDirectoryInfo other)
+        {
+            return other != null && Name == other.Name && Path == other.Path;
+        }
```

Comment:
Are you sure these still work? These methods work if a `CustomDataDirectoryInfo` just contains a `Name` and a `Path`, but clearly, you've added a lot more stuff, so these methods should be updated to reflect them.

Reply:
The most comparisons should still work, because path is unique to each `CustomDataDirectoryInfo`, but I will take a look and update them if necessary.

Reply:
This looks promising! Thank you @JoschiGrey and @DelnarErsike for this!